manim/utils/bezier.py

```diff
@@ -43,34 +54,39 @@ def bezier(
 
     Returns
     -------
-    typing.Callable[[float], typing.Union[int, typing.Iterable]]
         function describing the bezier curve.
+        You can pass a t value between 0 and 1 to get the corresponding point on the curve.
     """
     n = len(points) - 1
-
     # Cubic Bezier curve
     if n == 3:
-        return (
-            lambda t: (1 - t) ** 3 * points[0]
+        return lambda t: np.asarray(
+            (1 - t) ** 3 * points[0]
             + 3 * t * (1 - t) ** 2 * points[1]
             + 3 * (1 - t) * t**2 * points[2]
-            + t**3 * points[3]
+            + t**3 * points[3],
+            dtype=PointDType,
         )
     # Quadratic Bezier curve
     if n == 2:
-        return (
-            lambda t: (1 - t) ** 2 * points[0]
-            + 2 * t * (1 - t) * points[1]
-            + t**2 * points[2]
+        return lambda t: np.asarray(
+            (1 - t) ** 2 * points[0] + 2 * t * (1 - t) * points[1] + t**2 * points[2],
+            dtype=PointDType,
         )
 
-    return lambda t: sum(
-        ((1 - t) ** (n - k)) * (t**k) * choose(n, k) * point
-        for k, point in enumerate(points)
+    return lambda t: np.asarray(
+        np.asarray(
+            [
+                (((1 - t) ** (n - k)) * (t**k) * choose(n, k) * point)
+                for k, point in enumerate(points)
+            ],
+            dtype=PointDType,
+        ).sum(axis=0)
     )
 
 
-def partial_bezier_points(points: np.ndarray, a: float, b: float) -> np.ndarray:
+# !TODO: This function has still a weird implementation with the overlapping points
```

Comment:
I'll let @MrDiver open an issue as I don't really know what it is about. Comment removed.

Reply:
This is about the implementation that it doesn't really do a good job of interpolating which causes a lot of weird behavior down the line because there is not smooth shrinking of the mesh or just cutting of because it keeps the amount of curves the same as far as i remember. I already fixed that implementation in https://github.com/ManimCommunity/manim/pull/2737
Or at least there i noticed and then remembered it again but it will be changed in the future anyway so no ticket needed.